packages/@cdktf/provider-generator/lib/get/constructs-maker.ts

```diff
@@ -236,13 +236,12 @@ export class ConstructsMaker {
         (target) => target instanceof ConstructsMakerProviderTarget
       ) as ConstructsMakerProviderTarget[];
 
-    if (providerTargets.length > 0) {
-      new TerraformProviderGenerator(
-        this.code,
-        schema.providerSchema,
-        providerTargets
-      );
-    }
+    providerTargets.forEach(
+      (provider) =>
+        new TerraformProviderGenerator(this.code, schema.providerSchema, [
```

Comment:
We should take a look if there is another case where this class is used. If not we should change the interface so it takes a single provider instead of the list of providers it does now